plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -368,7 +368,9 @@ export class NunjucksWorkflowRunner implements WorkflowRunner {
             try {
               let prevValue: U | undefined;
               if (prevTaskState) {
-                const prevState = (prevTaskState.state as TaskState)?.[key];
+                const prevState = (
+                  prevTaskState.state?.checkpoints as TaskState
```

Comment:
Think that `TaskState` in this case should be renamed to `CheckpointState` or something right now? Feels a little strange to call this `TaskState` now as it's not really a task

Reply:
Haha, I just did that change few minutes ago :)

Reply:
Do you prefer to have it as CheckpointState`? I named it `CheckpointsState`

Reply:
Well it depends. `CheckpointsState` would be the type of `state.checkpoints` but `CheckpointState` would be the type of `state.checkpoints.${id}` right? Probably don't need both, but I think it's better to use the narrower type of `CheckpointState` and just type `{[key: string}: CheckpointState}` for the `state.checkpoints`